lib/fastlane/actions/gym.rb

```diff
@@ -17,6 +17,8 @@ def self.run(values)
             values[:provisioning_profile_path] = File.expand_path(sigh_path) if sigh_path
           end
 
+          SetBuildNumberRepositoryAction.add_build_number_build_setting(values, :xcargs)
```

Comment:
Enforcing to set a build number in gym action is not a good idea. Myself, I always want to control this in my flow.